gtest/src/program.rs

```diff
@@ -88,6 +88,7 @@ pub trait WasmProgram: Debug {
     fn init(&mut self, payload: Vec<u8>) -> Result<Option<Vec<u8>>, &'static str>;
     fn handle(&mut self, payload: Vec<u8>) -> Result<Option<Vec<u8>>, &'static str>;
     fn handle_reply(&mut self, payload: Vec<u8>) -> Result<Option<Vec<u8>>, &'static str>;
+    fn handle_signal(&mut self, payload: Vec<u8>) -> Result<Option<Vec<u8>>, &'static str>;
```

Comment:
atm we will deprecate anything except `gr_reply_to` and `gr_exit_code` for `handle_signal` entry, so there should be `Result<(), &'static str` and backend requires changes in checking context of execution as it did for mentioned above syscalls